packages/tree/src/Tree.tsx

```diff
@@ -0,0 +1,373 @@
+  const [isReady, setIsReady] = useState(false);
+
+  const walkTree = useCallback((item: TreeItem, fn: (item: TreeItem) => void) => {
+    fn(item);
+    if (item.children) {
+      for (const child of item.children) {
+        walkTree(child, fn);
+      }
+    }
+  }, []);
+
+  useEffect(() => {
+    if (treeItems) {
+      if (!isReady) {
+        for (const item of treeItems) {
+          item.isExpanded = item.isExpanded || expandAll || false;
+          item.isHidden = item.isHidden || false;
+          item.isDisabled = item.isDisabled || false;
+          item.isSelected = selectedItems.map((item) => item.id).includes(item.id) || item.isSelected || false;
+          item.areAllChildrenSelected = areAllChildrenSelected(item);
+        }
+      }
+      setIsReady(true);
+    }
+  }, [treeItems, expandAll, isReady, selectedItems]);
+
+  const handleItemSelected = useCallback(() => {
+    if (onItemsSelected) {
+      onItemsSelected(selectedList);
+    }
+  }, [selectedList, onItemsSelected]);
+
+  useEffect(() => {
+    if (areAllSelected(treeItems)) {
+      setRootSelectText(DESELECT_ALL);
+    } else {
+      setRootSelectText(SELECT_ALL);
+    }
+
+    for (const item of treeItems) {
+      item.areAllChildrenSelected = areAllChildrenSelected(item);
+    }
+
+    handleItemSelected();
+  }, [selectedList, treeItems, areAllSelected, handleItemSelected]);
+
+  useEffect(() => {
+    if (selectedItems.length > 0) {
+      setSelectedList(selectedItems);
+      for (const item of treeItems) {
+        walkTree(item, (item) => {
+          item.areAllChildrenSelected = areAllChildrenSelected(item);
+        });
+      }
+    }
+  }, [selectedItems, walkTree, treeItems]);
+
+  const filterItems = (items: TreeItem[], searchTerm: string) => {
+    const searchValue = searchTerm.toUpperCase();
+    const rootItems = items.filter((item) => item.name.toUpperCase().includes(searchValue));
+    for (const item of rootItems) item.isHidden = false;
+
+    const hiddenItems = items.filter((item) => !rootItems.map((rootItem) => rootItem.id).includes(item.id));
+    for (const hiddenItem of hiddenItems) hiddenItem.isHidden = true;
+    for (const item of items) {
+      if (item.children) {
+        const matchedChildren = item.children.filter((item) => item.name.toUpperCase().includes(searchValue));
+        if (matchedChildren?.length > 0) {
+          item.isExpanded = true;
+        }
+
+        filterItems(item.children, searchTerm);
+      }
+    }
+  };
+
+  const toggleSelect = (item: TreeItem) => {
+    const itemToUpdate = treeItems.find((t) => t.id === item.id);
+    if (itemToUpdate) {
+      item.isSelected = !item.isSelected;
+      if (item.isSelected) {
+        setSelectedList([...selectedList, item]);
+      } else {
+        setSelectedList(selectedList.filter((item) => item.id !== itemToUpdate.id));
+      }
+    }
+    setTreeItems([...treeItems]);
```

Comment:
What is this line used for?  I don't see where `treeItems` changed

Reply:
It doesn't rerender without this, pretty sure that is why I had to do this.

Reply:
I reran this today removing the setTreeItems and it seems to show the tree items as selected just fine. When I built this originally a few months ago I do remember having some issues, but I don't see it now. Removed.